app.js

```diff
@@ -1 +1,165 @@
+}
+
+/**
+picks 3 images and returns them, no duplicates. ignores images from the previous round.
+**/
+function randomImages(remainingImages, previousImages) {
+  var chosenIndex;
+  var chosenImages = [];
+  var numToChoose = 3;
+  for (var i = 0; i < numToChoose; i++) {
+    chosenIndex = randomIndex(remainingImages);
+    chosenImages.push(remainingImages[chosenIndex]);
+    remainingImages.splice(chosenIndex, 1); // removes chosen item
+  }
+  for (var j = 0; j < previousImages.length; j++) {
+    remainingImages.push(previousImages[j]); // add back in the previous images
+  }
+  return chosenImages;
+}
+
+/**
+sticks the chosen images in a div to print. adds event listeners to each.
+**/
+function renderImages(chosenImages) {
+  var div = document.createElement('div');
+  div.id = 'app-images';
+  var image;
+  var printImage;
+  for (var i = 0; i < chosenImages.length; i++) {
+    printImage = chosenImages[i];
+    image = document.createElement('img');
+    image.src = printImage.imagePath; //display the image
+    image.id = printImage.imageId;
+    printImage.timesShow++; //increment times shown
+    image.addEventListener('click', handleImageClick);
+    div.appendChild(image);
+  }
+  return div;
+}
+
+/**
+handles a click on one of the set of images. gives another set if more rounds required, removes the listener if done.
+**/
+function handleImageClick(e) {
+  console.log(currentRound);
+  var imageId = e.target.id;
+  var imageObj;
+  for (var i = 0; i < allImages.length; i++) {
+    if (imageId === allImages[i].imageId) {
+      imageObj = allImages[i];
+      i = allImages.length;
+    }
+  }
+  imageObj.timesClick++;
+  if (currentRound != maxRound) {
+    previousImages = randomImages(remainingImages, previousImages);
+    imageBox.removeChild(document.getElementById('app-images'));
+    imageBox.appendChild(renderImages(previousImages));
+  } else {
+    console.log('DONE!');
+    removeAllListeners();
+    imageBox.appendChild(renderStatistics());
+  }
+  currentRound++;
+}
+
+/**
+removes the click event listeners from all the images
+**/
+function removeAllListeners() {
+  var imageElements = document.querySelectorAll('#app-images img');
+  for (var i = 0; i < imageElements.length; i++) {
+    imageElements[i].removeEventListener('click', handleImageClick);
+  }
+}
+
+/**
+renders the times clicked for every image in a list
+**/
+function renderStatistics() {
+  var statsList = document.createElement('ul');
+  var imageStats;
+  for (var i = 0; i < allImages.length; i++) {
+    imageStats = document.createElement('li');
+    imageStats.textContent = allImages[i].timesClick + ' votes for the ' + allImages[i].imageTitle;
+    statsList.appendChild(imageStats);
+  }
+  return statsList;
+}
+
+/**
+MAIN APPLICATION
+**/
+
+var remainingImages = allImages.slice(); // copy to leave original alone
+
+var previousImages = [];
+var maxRound = 25;
+var currentRound = 1;
+var imageBox = document.getElementById('app');
```

Comment:
Move var declarations / definitions to the top of the file.   This is common convention.